app/src/androidTest/java/org/simple/clinic/overdue/AppointmentRepositoryAndroidTest.kt

```diff
@@ -775,4 +775,54 @@ class AppointmentRepositoryAndroidTest {
     val (appointment) = appointmentRepository.lastCreatedAppointmentForPatient(patientId).blockingFirst()
     assertThat(appointment!!).isEqualTo(secondAppointment)
   }
+
+  @Test
+  fun marking_appointment_older_than_current_date_should_work_correctly() {
+    val patientId = UUID.randomUUID()
+    val appointmentUuid = UUID.randomUUID()
+    val scheduleDate = LocalDate.now(testClock).plusMonths(1)
+
+    testClock.advanceBy(Duration.ofHours(1))
+    database
+        .appointmentDao()
+        .save(listOf(testData.appointment(
+            uuid = appointmentUuid,
+            patientUuid = patientId,
+            status = SCHEDULED,
+            syncStatus = SyncStatus.DONE,
+            scheduledDate = scheduleDate,
+            createdAt = Instant.now(testClock),
+            updatedAt = Instant.now(testClock)
+        )))
+
+    val firstAppointment = database.appointmentDao().getOne(appointmentUuid)
+    testClock.advanceBy(Duration.ofHours(1))
+
+    appointmentRepository.markAppointmentsCreatedBeforeTodayAsVisited(patientId).blockingAwait()
+
+    val lastAppointment = database.appointmentDao().getOne(appointmentUuid)!!
+    assertThat(lastAppointment).isEqualTo(firstAppointment)
+
+    testClock.advanceBy(Duration.ofDays(1))
+
+    database
+        .appointmentDao()
+        .save(listOf(testData.appointment(
+            patientUuid = patientId,
+            scheduledDate = scheduleDate,
+            status = SCHEDULED,
+            syncStatus = SyncStatus.PENDING,
+            createdAt = Instant.now(testClock),
+            updatedAt = Instant.now(testClock)
+        )))
+    appointmentRepository.markAppointmentsCreatedBeforeTodayAsVisited(patientId).blockingAwait()
+
+    val oldAppointment = database.appointmentDao().getOne(lastAppointment.uuid)
+
+    oldAppointment?.apply {
```

Comment:
Instead of saving this as a separate `oldAppointment`, why can't we just inline it?

```kotlin
database.appointmentDao().getOne(appointment.uuid)!!.let { oldAppointment ->
  // Run assertions
}
```